test/test_testing.py

```diff
@@ -2751,6 +2751,19 @@ def _test_push_and_pull_data(self, original_data, encoding, stream_class):
             received_data = infile.read()
         self.assertEqual(original_data, received_data)
 
+    def test_push_and_pull_larger_file(self):
```

Comment:
Nice. Would be great if while you're at it you can add a couple of tests to `TestRealPebble` so push and pull can be easily tested against the real Pebble server too.

Reply:
So, here's an interesting bit (and the reason why this MR includes a bunch of seemingly unrelated stuff from #645): this test was added to a test mixin which is used for testing both against the mock Pebble backend and the real Pebble.

Actually those tests in #645 are something that's worth perhaps some extra thought.  They're in the test_testing module, but they also test real pebble.  I kind of feel like the overall "domain" is pebble, and they should be moved to test_pebble.py.